### PR TITLE
Feature/learning rate decay

### PR DESCRIPTION
**Pull request by davis0378**
我加上了學習率衰減，會在訓練時依據epoch來逐漸讓學習率變小

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -71,6 +73,9 @@ def __init__(self, input_dim, hidden_dim, output_dim, learning_rate=0.01, reg_la
         self.bc = np.zeros((hidden_dim, 1))
         self.by = np.zeros((output_dim, 1))
 
+    def learning_rate_decay(self, epoch):
+        self.learning_rate = self.init_learning_rate / (1 + self.decay_factor * epoch)
```
Comment (sharkfoolish) [inline main.py:77]: 他不是有很多種衰減的公式，啊你為什麼選這個？

Comment (davis0378) [reply main.py:77]: 我試過定期衰減、指數衰減、1/t 衰減，其中1/t 衰減在計算上較方便，而且不會一下子下降到很多

Comment (sharkfoolish) [reply main.py:77]: OK 那就用1/t 衰減

**Review by seafarer0675 — APPROVED** [review]

**Review by sharkfoolish — APPROVED** [review]
